Examples/MagicWeather/MagicWeather/Sources/Controllers/UserViewController.swift

```diff
@@ -0,0 +1,122 @@
+ Configured in /Resources/UI/Main.storyboard
+ */
+
+class UserViewController: UIViewController {
+
+    @IBOutlet var userIdLabel: UILabel!
+    @IBOutlet var statusLabel: UILabel!
+
+    override func viewDidAppear(_ animated: Bool) {
+        super.viewDidAppear(animated)
+        
+        /// - Refresh details when the User tab is viewed
+        refreshUserDetails()
+    }
+
+    func refreshUserDetails() {
+        self.userIdLabel.text = Purchases.shared.appUserID
+        
+        if Purchases.shared.isAnonymous {
+            self.navigationItem.rightBarButtonItem = UIBarButtonItem(title: "Log In", style: .done, target: self, action: #selector(presentLoginDialog))
+        } else {
+            self.navigationItem.rightBarButtonItem = UIBarButtonItem(title: "Log Out", style: .done, target: self, action: #selector(logout))
+        }
+        
+        Purchases.shared.purchaserInfo { (purchaserInfo, error) in
+            if purchaserInfo?.entitlements[Constants.entitlementID]?.isActive == true {
+                self.statusLabel.text = "Active"
+                self.statusLabel.textColor = .green
+            } else {
+                self.statusLabel.text = "Not Active"
+                self.statusLabel.textColor = .red
+            }
+        }
+    }
+}
+
+/*
+ How to login and identify your users with the Purchases SDK.
+ 
+ These functions mimic displaying a login dialog, identifying the user, then logging out later.
+ 
+ Read more about Identifying Users here: https://docs.revenuecat.com/docs/user-ids
+ */
+extension UserViewController {
+    
+    /// - Login method. Replace this with your own login sequence.
+    @objc
+    func presentLoginDialog() {
+        let alert = UIAlertController(title: "Log In", message: "Enter your username.", preferredStyle: .alert)
+        alert.addTextField { (textField) in
+            textField.placeholder = "Username"
+        }
+        alert.addAction(UIAlertAction(title: "Cancel", style: .cancel, handler: nil))
+        alert.addAction(UIAlertAction(title: "Log In", style: .default, handler: { (action) in
+            /// - Ensure a username was entered
+            if let username = alert.textFields?.first?.text, username.isEmpty == false {
+                
+                #warning("Public-facing usernames aren't optimal for user ID's - you should use something non-guessable, like a non-public database ID. For more information, visit https://docs.revenuecat.com/docs/user-ids.")
+                /// - Call `identify` with the Purchases SDK with the unique user ID
+                Purchases.shared.identify(username) { (purchaserInfo, error) in
+                    if let error = error {
+                        self.present(UIAlertController.errorAlert(message: error.localizedDescription), animated: true, completion: nil)
+                    }
+                    
+                    self.refreshUserDetails()
```

Comment:
very minor, but you don't really need to refresh user details if there's an error, so this could be an else branch